src/diagnose.ts

```diff
@@ -34,7 +35,7 @@ export class DiagnoseTool {
   #extension: Extension
 
   constructor() {
-    this.#config = new Configuration({})
+    this.#config = this.getConfigObject()
```

Comment:
Calling `this.getConfigObject()` will in turn call `require(Configuration.clientFilePath)`, right? And it's for this require of the client file that we need `_APPSIGNAL_DIAGNOSE = true`, so that the client that we're requiring doesn't start the extension on initialisation.

But when the constructor is called here, `_APPSIGNAL_DIAGNOSE` hasn't been set yet -- that happens later, in the `generate()` method in line 48 of this file. So when the client file is required, the environment variable will not be set yet, and the extension will be started.

I think it would make sense to set the environment variable inside `getConfigObject` instead, also for readability -- that way the environment variable is set in the context where it's needed.